core/state/intra_block_state.go

```diff
@@ -155,22 +155,28 @@ func (sdb *IntraBlockState) Reset() {
 	//clear(sdb.stateObjects)
 	sdb.stateObjectsDirty = make(map[libcommon.Address]struct{})
 	//clear(sdb.stateObjectsDirty)
-	sdb.logs = make(map[int][]*types.Log)
+	sdb.logs = []types.Logs{}
```

Comment:
You can use
```
sdb.logs = sdb.logs[:0]
```
instead.
That way you preserve the slice' capacity and thus can drop `logSize` entirely in favour of `len(logs)`.

Reply:
done

Reply:
Sorry, I initially misunderstood the logic here. Perhaps it's actually preferable to do
```
sdb.logs = []types.Logs{}
```
instead of 
```
sdb.logs = sdb.logs[:0]
```
in order to release memory. Hard to tell without profiling.